File: InsertionSort/InsertionSort.java
import org.jetbrains.annotations.NotNull;

public class InsertionSort {
    public static void insertionSort(int @NotNull [] array) {
        for(int i = 0; i < array.length; i++) {
            for(int j = i; j > 0; j--) {
                if(array[j] < array[j - 1]) {
                    int temp = array[j];
                    array[j] = array[j - 1];
                    array[j - 1] = temp;
                }
                else break;
            }
        }
    }

    public static void printArray(String arrayName, int @NotNull [] array) {
        System.out.print(arrayName + " = { ");
        for(int i = 0; i < array.length; i++) {
            System.out.print(array[i]);
            if(i != array.length - 1) System.out.print(", ");
        }
        System.out.println(" }");
    }

    public static void main(String[] args) {
        int[] numberArray = {92, 8, 73, 55, 22, 12, 37, 83};
        printArray("UNSORTED ARRAY", numberArray);
        // UNSORTED ARRAY = { 92, 8, 73, 55, 22, 12, 37, 83 }
        insertionSort(numberArray);
        printArray("SORTED ARRAY", numberArray);
        // SORTED ARRAY = { 8, 12, 22, 37, 55, 73, 83, 92 }
    }
}

File: InsertionSort/insertionSort.c
#include <stdio.h>

void insertionSort(int array[], int arraySize) {
    for(int i = 0; i < arraySize; i++) {
        for(int j = i; j > 0; j--) {
            if(array[j] < array[j - 1]) {
                int temp = array[j];
                array[j] = array[j - 1];
                array[j - 1] = temp;
            }
            else break;
        }
    }
}

void printArray(char arrayName[], int array[], int arraySize) {
    printf("%s = { ", arrayName);
    for(int i = 0; i < arraySize; i++) {
        printf("%d", array[i]);
        if(i != arraySize - 1) printf(", ");
    }
    printf(" }\n");
}

int main() {
    int numberArray[] = {9, 100, 2, 28, 25, 1, 48, 14};
    int arraySize = sizeof(numberArray) / sizeof(int);
    printArray("UNSORTED ARRAY", numberArray, arraySize);
    // UNSORTED ARRAY = { 9, 100, 2, 28, 25, 1, 48, 14 }
    insertionSort(numberArray, arraySize);
    printArray("SORTED ARRAY", numberArray, arraySize);
    // SORTED ARRAY = { 1, 2, 9, 14, 25, 28, 48, 100 }
    return 0;
}
